source/logs/custom-entries.txt

```diff
@@ -0,0 +1,125 @@
+.. _mongosh-custom-log-entries:
+
+========================
+Write Custom Log Entries
+========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+You can write custom :ref:`MongoDB Shell log entries <mdb-shell-logs>`
+from your :ref:`MongoDB Shell scripts <mdb-shell-write-scripts>`. Custom
+log entries help with debugging, error handling, and alert you when the
+script performs specific functions.
+
+About this Task
+---------------
+
+MongoDB Shell supports the following log levels for custom log entries:
+
+- ``log.debug()``
+- ``log.error()``
+- ``log.fatal()``
+- ``log.info()``
+- ``log.warn()``
+
+Steps
+-----
+
+.. procedure::
+
+   .. step:: Create a script that writes a custom log entry
+
+      The following script inserts documents into the ``movies``
+      collection and writes a custom ``info`` log entry. If the script
+      errors, it writes a custom ``error`` log entry instead.
+
+      .. code-block:: javascript
+         :emphasize-lines: 22,24
+
+         // connect-and-insert-with-log-entry.js
+         
+         try {
+            db = connect( 'mongodb://localhost/myDatabase' );
+
+            db.movies.insertMany( [
+               {
+                  title: 'Titanic',
+                  year: 1997,
+                  genres: [ 'Drama', 'Romance' ]
+               },
+               {
+                  title: 'Spirited Away',
+                  year: 2001,
+                  genres: ['Animation', 'Adventure', 'Family' ]
+               },
+               {
+                  title: 'Casablanca',
+                  genres: [ 'Drama', 'Romance', 'War' ]
+               }
+            ] )
+            log.info('InsertData: [$(new Date().toISOString()}] Inserted 3 movies')
+         } catch (error) {
+            log.error('ERROR: [$(new Date().toISOString()}] Insert failed. Reason: ${error.message}');
```

Comment:
I think the syntax here contains unbalanced parentheses – are these intended to be template strings?

We'd probably want a slightly better format here anyway:
- There's no need to include a timestamp, as the logger will automatically include one already
- Attributes like error messages can be passed explicitly, rather than being made part of the message itself

e.g.

```suggestion
            log.error('Insert failed', { error: error.message });
```

Reply:
Thanks Anna! Will update this.